pyproject.toml

```diff
@@ -36,6 +36,7 @@ dependencies = [
     "attrs~=25.1",
     "colorlog~=6.9",
     "multidict~=6.1",
+    "typing_extensions~=4.4",
```

Comment:
Instead of depending on `typing_extensions`, we can instead use a shim: https://github.com/hikari-py/hikari/pull/2239/files#diff-9e76acd14a9e45dafc4708fa9e301dd04a19f363acbce41f6c2a73930bc34cfb

Reply:
I don't like re-implementing lang stuff which might change or be added to (even if it's just stuff like the attributes a decorator adds) as i'll likely miss that when adding new python versions but if that's what you want i can change this

Reply:
An example would be that the warnings.deprecated shim in `hikari.internal.deprecation` has been failing to set the `.__deprecated__` attribute since the PEP changed

Reply:
Can I copy over the shim from that ruff pr?

Reply:
> I don't like re-implementing lang stuff which might change or be added to (even if it's just stuff like the attributes a decorator adds) as i'll likely miss that when adding new python versions but if that's what you want i can change this.
An example would be that the warnings.deprecated shim in hikari.internal.deprecation has been failing to set the .__deprecated__ attribute since the PEP changed

Thats mostly runtime, so I wouldnt really worry about it. We only want `override` for its typehinting capabilities.

> Can I copy over the shim from that ruff pr?

Absolutely! I dont use there anymore, so it can be added as part of this pr and I will remove it from there

Reply:
Note that for mypy to not complain (nor ruff in the future) you will have to use it like this
```
if typing.TYPE_CHECKING:
	import typing_extensions
else:
	from hikari.internal import typing_extensions
```